File: skill_sdk/tests/fixtures/js-project/src/core/ProcessorConfig.js
/**
 * Configuration for processors. Mirrors ProcessorConfig in the Java fixture.
 */
export class ProcessorConfig {
  /**
   * @param {number} timeout
   * @param {number} retries
   */
  constructor(timeout, retries) {
    /** @type {number} */
    this._timeout = timeout;
    /** @type {number} */
    this._retries = retries;
  }

  /**
   * @returns {number}
   */
  getTimeout() {
    return this._timeout;
  }

  /**
   * @returns {number}
   */
  getRetries() {
    return this._retries;
  }
}

File: skill_sdk/tests/fixtures/java-project/src/main/java/com/fixture/core/ProcessorConfig.java
package com.fixture.core;

/**
 * Configuration for processors. Mirrors ProcessorConfig struct in main.go.
 */
public class ProcessorConfig {

    private final int timeout;
    private final int retries;

    public ProcessorConfig(int timeout, int retries) {
        this.timeout = timeout;
        this.retries = retries;
    }

    public int getTimeout() {
        return timeout;
    }

    public int getRetries() {
        return retries;
    }
}
